spec/system/admin_manage_auth_settings_spec.rb

```diff
@@ -48,4 +53,42 @@
     expect(auth_settings.enable_partial_email_signup).to be(false)
     expect(auth_settings.enable_partial_sms_signup).to be(true)
   end
+
+  it "shows a warning message when the SMS gateway service is not configured" do
+    Decidim.configure do |config|
+      config.sms_gateway_service = nil
+    end
+    click_link "Authentication settings"
+    expect(page).to have_current_path(decidim_half_signup_admin.edit_auth_setting_path(slug: "authentication_settings"))
+    expect(page).to have_css(".is-active", text: "Authentication settings")
+    expect(page.find("#auth_setting_enable_partial_sms_signup")).to be_disabled
+    expect(page).to have_content("This option is disabled please contact the host of the platform to enable it.")
+  end
+
+  it "shows an error message when the SMS gateway service is not configured if the user tries to force the change" do
+    click_link "Authentication settings"
+    expect(page).to have_current_path(decidim_half_signup_admin.edit_auth_setting_path(slug: "authentication_settings"))
+    expect(page).to have_css(".is-active", text: "Authentication settings")
+    check "Enable partial sign up and sign in using SMS verification"
+    check "Enable partial sign up and sign in using email verification"
+
+    Decidim.configure do |config|
```

Comment:
Prefer to define mock for `Decidim.configure.sms_gateway_service` 

Example : `allow(Decidim.config.sms_gateway_service).to receive(:present?).and_return(true)`